main/src/main/scala/sbt/internal/ClassLoaders.scala

```diff
@@ -146,8 +146,9 @@ private[sbt] object ClassLoaders {
         val allTestDependencies = if (layerTestDependencies) allDependenciesSet else Set.empty[File]
         val allRuntimeDependencies = (if (layerDependencies) rawRuntimeDependencies else Nil).toSet
 
+        val scalaLibrarySet = Set(si.libraryJar)
```

Comment:
Why Set vs. List?

Reply:
I guess because `testDependencySet` etc are also sets, and you can use it later as `fullCP.filterNot(testDependencySet ++ runtimeDependencies ++ scalaLibrarySet)`, but this could go either way.